bittensor/utils/__init__.py

```diff
@@ -669,43 +672,55 @@ def update_curr_block(block_number: int, block_bytes: bytes, diff: int, lock: mu
 
                 update_curr_block(block_number, block_bytes, difficulty, check_block)
                 # Set new block events for each solver
-                for w in solvers:
-                    w.newBlockEvent.set()
-                    
-            # Get times for each solver
-            time_total = 0
+
+                for worker in solvers:
+                    worker.newBlockEvent.set()
+
             num_time = 0
+            # Get times for each solver
             for _ in solvers:
                 try:
-                    time_ = time_queue.get_nowait()
-                    time_total += time_
+                    _ = time_queue.get_nowait()
                     num_time += 1
 
                 except Empty:
                     break
             
+            time_since = time.time() - start_time
             if num_time > 0:
-                time_avg = time_total / num_time
-                itrs_per_sec = TPB*update_interval*num_processes / time_avg
-                time_since = time.time() - start_time
+                time_since_last = time.time() - time_last
```

Comment:
you might run into inconsistencies here. calling time.time() and then one more time two lines later might give you a millisecond difference. Might be better to give it to a variable and use that instead. 

```
time_now = time.time()
time_since = time_now - start_time

if num_time > 0:
     time_since_last = time_now - time_last
```